pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/Pql2Compiler.java

```diff
@@ -42,6 +47,14 @@ public BrokerRequest compileToBrokerRequest(String expression) throws Pql2Compil
       CharStream charStream = new ANTLRInputStream(expression);
       PQL2Lexer lexer = new PQL2Lexer(charStream);
       lexer.setTokenFactory(new CommonTokenFactory(true));
+      lexer.removeErrorListeners();
+      lexer.addErrorListener(new BaseErrorListener() {
+        @Override
+        public void syntaxError(@NotNull Recognizer<?, ?> recognizer, @Nullable Object offendingSymbol, int line,
+            int charPositionInLine, @NotNull String msg, @Nullable RecognitionException e) {
+          throw new Pql2CompilationException(msg);
```

Comment:
can we add more information in the exception message - exact fragment with error  using the offendingSymbol, line, charPositionInLine